pkg/frontend/openshiftcluster_preflightvalidation.go

```diff
@@ -0,0 +1,137 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+)
+
+// /subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{resourceProviderNamespace}/deployments/{deploymentName}/preflight?api-version={api-version}
+func (f *frontend) preflightValidation(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+
+	var header http.Header
+	var b []byte
+
+	body := r.Context().Value(middleware.ContextKeyBody).([]byte)
+
+	resources, err := unmarshalRequest(body)
+	if err == nil {
+		resourceCount := len(resources.Resources)
+		ch := make(chan api.ValidationResult, resourceCount)
+		for _, raw := range resources.Resources {
+			// get typeMeta from the raw data
+			typeMeta := &api.ResourceTypeMeta{}
```

Comment:
```suggestion
			typeMeta := api.ResourceTypeMeta{}
```

since you use pointer in the unmarshal below, this line should be a struct